### PR TITLE
Site monitoring: Enable loading skeleton when changing tab from metrics to log

### PR DESCRIPTION
**Pull request by sejas**
- Closes https://github.com/Automattic/dotcom-forge/issues/3832

## Proposed Changes

* Add skeleton loading to tables

## Testing Instructions

* On an atomic site, access `/site-monitoring/{siteSlug}`
* Click on `PHP Logs` or `Web Server Logs`
* Observe the loading skeleton for a few seconds

### Screencast

https://github.com/Automattic/wp-calypso/assets/779993/5792333f-74e1-4971-936e-77afeb48aab3

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

### CODE REVIEW

File: client/my-sites/site-monitoring/components/site-logs-table/index.tsx [hunk 1]
```diff
@@ -51,7 +51,7 @@ export const SiteLogsTable = memo( function SiteLogsTable( {
 		return generateRowKeys( logs );
 	}, [ logs ] );
 
-	const [ latestLogType, setLatestLogType ] = useState( logType );
+	const [ latestLogType, setLatestLogType ] = useState< LogType | undefined | null >( null );
```
Comment (danielbachhuber) [inline client/my-sites/site-monitoring/components/site-logs-table/index.tsx:54]: Because I'm a TypeScript/JavaScript newbie, how does this fix the issue?

**Review by danielbachhuber — APPROVED** [review]
Works great! 🎉

**Review by wojtekn — APPROVED** [review]
It fixes the issue for me.

File: client/my-sites/site-monitoring/components/site-logs-table/index.tsx (hunk 1 above)
Comment (sejas) [reply client/my-sites/site-monitoring/components/site-logs-table/index.tsx:54]: **Context**:
The log tables have two different loading states.
One that is a skeleton when it loads the first time or when you change between tabs.
And the second one which is a freeze-opacity when you change the pagination in the same table.

**Problem**:
The approach that I ported from Philip is that when you change between logs tab `logType`, we display the skeleton version of loading. We use `latestLogType` to track what was the previous `logType` value.

But when you came from the charts tab or when you were loading the logs directly. The component was initializing the `latestLogType` to the current `logType` in the first render which was skipping the loading state.

**Solution**:
To fix the problem, I changed the initial value of `latestLogType` to `null` with `useState( null )`. So we know we are mounting the component for the first time and we should display the skeleton.

The `<`, `>` on `useState< LogType | undefined | null >( null )` are used to define the type for the `latestLogType` variable.

Comment (sejas) [reply client/my-sites/site-monitoring/components/site-logs-table/index.tsx:54]: The simplest solutions are my favorite ones ❤️

**Pull request merged**